modules/gateways/callback/blockonomics.php

```diff
@@ -80,7 +81,10 @@
 } else {
 	$paymentAmount = $order['value'];
 }
-
+$exchangerate = Capsule::table('tblcurrencies')
+->where('code', $order['order_currency'])
+->get();
+$paymentAmount = $paymentAmount / $exchangerate[0]->rate;
```

Comment:
$paymentAmount should be rounded to 2 decimal places. Not doing so causes the correct amount to be received but the order status does not change to PAID.

This design requires re-evaluation. The exchange rate may be required to be saved during checkout to ensure this does not change between the time of making the payment and the payment being confirmed and added to the order.

Reply:
>This design requires re-evaluation. The exchange rate may be required to be saved during checkout to ensure this does not change between the time of making the payment and the payment being confirmed and added to the order.
This should be fine. Exchange rate don't change so much. So need to do extra effort here

>$exchangerate[0]->rate
Is this correct code ? Will it work if client currency is EUR and converttoForProcessing in not selected. In that case won't it take USD/EUR exchange rate incorrectly ?